### PR TITLE
Implement multi-day holidays

### LINKED ISSUE #1911: [TW-1898] support holidays longer then 1 day
_Richard Brinkman on 2017-03-02T13:12:14Z says:_ 

 Currently it is not possible to enter holidays that are longer than 1 day. It would be nice to have some sort of range.

Say:
holiday.en-US1.name=Eastern
holiday.en-US1.start=20150403
holiday.en-US1.end=20150406

instead of:
holiday.en-US1.name=Good friday
holiday.en-US1.date=20150403
holiday.en-US2.name=Eastern
holiday.en-US2.date=20150405
holiday.en-US3.name=Eastern monday
holiday.en-US3.date=20150406

For longer holidays like a summer vacation it is just tedious to enter all the individual dates.

### PR DESCRIPTION
**Pull request by DanielMowitz**
#### Description

This PR resolves #1911 and cleans up the code for holiday colorization.

#### Additional information...

- [x] Have you run the test suite?
Output of test/problems:
```errors
Failed:
template.t                          2
version.t                           1

Unexpected successes:

Skipped:
dependencies.t                      1
export.t                            1
feature.default.project.t           4
hooks.on-modify.t                   1
import.t                            1
recurrence.t                        1
template.t                          3
tw-1999.t                           2
wait.t                              1

Expected failures:
dependencies.t                      3
filter.t                            3
hyphenate.t                         1
lexer.t                             4
quotes.t                            1
template.t                          1
tw-2124.t                           1
```

### CODE REVIEW

File: src/commands/CmdCalendar.cpp [hunk 1]
```diff
@@ -544,13 +560,27 @@ std::string CmdCalendar::renderMonths (
...
                   cellColor.blend (color_holiday);
               }
+
+              if (hol.first.substr (hol.first.size () - 5) == "start" &&
+                  Context::getContext ().config.has ("holiday." + hol.first.substr (8, hol.first.size () - 14) + ".end"))
+              {
+                auto start = hol.second;
+                auto end = Context::getContext ().config.get ("holiday." + hol.first.substr (8, hol.first.size () - 14) + ".end");
+                Datetime holStart (start.c_str (), Context::getContext ().config.get ("dateformat.holiday"));
+                Datetime holEnd   (end.c_str (), Context::getContext ().config.get ("dateformat.holiday"));
```
Comment (tbabej) [inline src/commands/CmdCalendar.cpp:579]: Perhaps we could move `Context::getContext ().config.get ("dateformat.holiday")` into its own variable? Gets quite repetitive over the code. Would be a nice refactoring win coming for this PR :slightly_smiling_face:

File: src/commands/CmdCalendar.cpp [hunk 2]
```diff
@@ -544,13 +560,27 @@ std::string CmdCalendar::renderMonths (
...
+                if (holStart <= date &&
+                    date     <= holEnd)
+                  cellColor.blend (color_holiday);
+              }
+				}
```
Comment (tbabej) [inline src/commands/CmdCalendar.cpp]: This looks like runaway indentation (same with opening bracket).

Comment (tbabej) [reply src/commands/CmdCalendar.cpp]: Fixed in subsequent commit.

**Review by tbabej — CHANGES_REQUESTED** [review]
Looks good! I left suggestions for some minor changes in the code (mostly around readability), but most importantly, we'll need to:
* add a test or two to make sure this works (at least a sanity check verifying that holidays with start and end do not make task calendar crash)
* documentation changes in the relevant sections of man pages for `task` and `taskrc`

File: src/commands/CmdCalendar.cpp [hunk 3]
```diff
@@ -380,11 +380,27 @@ int CmdCalendar::execute (std::string& output)
           if (it.first.substr (it.first.size () - 4) == "name")
           {
             auto holName = Context::getContext ().config.get ("holiday." + it.first.substr (8, it.first.size () - 13) + ".name");
```
[the remark this answers is not in the document]
Comment (DanielMowitz) [reply src/commands/CmdCalendar.cpp]: You're right, this is a good place to start with that.

**Comment by DanielMowitz** [comment]
This should cover all requested changes. I'm not quite content with the comment on the tests though.

**Comment by tbabej** [comment]
Looks good to me. Thanks for the PR, @DanielMowitz !

**Pull request merged**